src/main/java/cc/aabss/eventutils/UpdateChecker.java

```diff
@@ -35,36 +36,52 @@ private void notifyUpdate(@NotNull String latestVersion) {
+                    .thenAccept(body -> {
+                        try {
+                            JsonObject latestVersionObj = JsonParser.parseString(body)
+                                    .getAsJsonArray()
+                                    .get(0)
+                                    .getAsJsonObject();
 
-            // Get latest version from Modrinth
-            final HttpClient httpClient = HttpClient.newHttpClient();
-            httpClient
-                    .sendAsync(HttpRequest.newBuilder()
-                                    .uri(new URI("https://api.modrinth.com/v2/project/alerts/version?game_versions=%5B%22" + Versions.MC_VERSION + "%22%5D"))
-                                    .header("User-Agent", "EventUtils/" + Versions.EU_VERSION + " (Minecraft/" + Versions.MC_VERSION + ")").build(),
-                            HttpResponse.BodyHandlers.ofString())
-                    .whenCompleteAsync((response, throwable) -> {
-                        if (throwable != null) {
-                            EventUtils.LOGGER.warn("Failed to check for updates", throwable);
-                            return;
-                        }
+                            if (latestVersionObj != null && latestVersionObj.has("version_number")) {
+                                final String latestVersion = latestVersionObj.get("version_number").getAsString();
+                                final String currentVersion = Versions.MC_VERSION + "-" + Versions.EU_VERSION;
 
-                        // Extract version
-                        final String latestVersion = JsonParser.parseString(response.body()).getAsJsonArray()
-                                .get(0).getAsJsonObject()
-                                .get("version_number").getAsString();
-                        if (latestVersion != null && !latestVersion.equals(Versions.MC_VERSION + "-" + Versions.EU_VERSION)) notifyUpdate(latestVersion);
-                        //? if java: >=21
-                        httpClient.close();
+                                if (!currentVersion.equals(latestVersion)) {
+                                    notifyUpdate(latestVersion);
+                                }
+                            } else {
+                                EventUtils.LOGGER.error("Failed to check for updates: Unexpected response from Modrinth");
+                            }
+                        } catch (Exception e) {
+                            EventUtils.LOGGER.error("Failed to parse update check:", e);
+                        }
+                    })
+                    .exceptionally(e -> {
+                        EventUtils.LOGGER.error("Failed to check for updates", e);
+                        return null;
                     });
-        } catch (final Exception e) {
-            EventUtils.LOGGER.warn("Failed to check for updates", e);
+        } catch (final URISyntaxException e) {
+            throw new RuntimeException(e);
```

Comment:
revert
should catch any exception just in-case
shouldnt throw runtimeexception as it crashes the game. we dont care if update checker fails so just silently log it